### PR TITLE
Wrap/Unwrap Token page & firsts tests

### PR DESCRIPTION
**Pull request by jmc171144**

### CODE REVIEW

File: src/Trakx.IndexManager.Client.Tests/WrappingTests.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        }
+
+        [Test]
+        public void test_click_on_wrap_or_unwrap_button_should_display_corresponding_choice_in_the_submit_button()
+        {
+            var cut = RenderComponent<WrapUnwrapToken>();
+            var wrapButton = cut.FindAll("button").ElementAt(0);
+            var unwrapButton = cut.FindAll("button").ElementAt(1);
+
+            wrapButton.Click();
```
[the remark this answers is not in the document]
Comment (jmc171144) [reply src/Trakx.IndexManager.Client.Tests/WrappingTests.cs:60]: This simulates a click on the "Wrap" button which allows the user to select the wrapping option

File: src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor [hunk 2]
```diff
@@ -1,7 +1,225 @@
...
+            "wBTC", "wLTC", "wXRP", "wXTZ"
+        };
+
+        /// <summary>
+        /// Returns an array of strings of the cryptocurrencies available for wrap/unwrap (according to the option selected upwind)
+        /// </summary>
+        /// <returns></returns>
+        public string[] CryptocurrencyOptions()
+        {
+            if (disabled == false)
```
Comment (monsieurleberre) [inline src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor:231]: reduce nesting with 
```
if(disabled) return new[] { "Error: select an option above" };
return wrapping ? wrappingOptions : unwrappingOptions;
```

File: src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor [hunk 3]
```diff
@@ -1,7 +1,225 @@
...
+        /// <summary>
+        /// Returns the correct address pattern (Regex) for the address FROM which the initial cryptocurrency is send.
+        /// </summary>
+        /// <returns>address from pattern</returns>
+        public string GetPatternFrom()
+        {
+            if (wrapping == false)
+            {
+                //if we unwrap, the to address is an eth address
+                return "^0x[a-fA-F0-9]{40}$";
```
Comment (monsieurleberre) [inline src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor:135]: use a private constant in the 2 methods, you can call it something explicit like
`ehtereumAddressPattern` then we know what you are doing

File: src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor [hunk 4]
```diff
@@ -1,7 +1,225 @@
...
+    </EditForm>
+</div>
+
+
+@code {
+
+    [Parameter] public EventCallback<MouseEventArgs> OnSubmitClick { get; set; }
+    [Parameter] public EventCallback<MouseEventArgs> OnResetClick { get; set; }
+
+    private Task HandleOnSubmitClick(MouseEventArgs args)
```
Comment (monsieurleberre) [inline src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor:61]: I would have thought you could directly bind to a method, without these event callbacks no?

File: src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor [hunk 5]
```diff
@@ -1,7 +1,234 @@
...
+@page "/Wrapping/wrap_unwrap_token"
+
+<h3>Wrap or Unwrap Tokens</h3>
+<br />
+
+<div>
+    <div class="box">
+        <div>
+            <div>Select an option below to wrap or unwrap tokens:<br /></div>
+            <div>
```
[the remark this answers is not in the document]
Comment (jmc171144) [reply src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor:14]: Yes I can use `<input type=radio>` ! but I can't use MatRadioButton or MatSlideToggle if I want to be able to test them (for now 😂)

File: src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor (hunk 4 above)
Comment (jmc171144) [reply src/Trakx.IndiceManager.Client/Pages/Wrapping/WrapUnwrapToken.razor:61]: No idea, documentations only use Event callbacks but I will look into

**Pull request merged**